crates/wast/src/resolve/component_names.rs

```diff
@@ -0,0 +1,817 @@
+use crate::ast::*;
+use crate::resolve::names::Namespace;
+use crate::Error;
+use std::convert::TryInto;
+use std::mem::replace;
+
+/// Resolve the fields of a component and everything nested within it, changing
+/// `Index::Id` to `Index::Num` and expanding alias syntax sugar.
+pub fn resolve<'a>(id: Option<Id<'a>>, fields: &mut Vec<ComponentField<'a>>) -> Result<(), Error> {
+    let mut resolve_stack = Vec::new();
+    resolve_fields(id, fields, &mut resolve_stack)
+}
+
+fn resolve_fields<'a, 'b>(
+    id: Option<Id<'a>>,
+    fields: &mut Vec<ComponentField<'a>>,
+    resolve_stack: &'b mut Vec<ComponentResolver<'a>>,
```

Comment:
This option seems to be passed around to most functions here, along with the redefinitions of `'a` and `'b`. Would you be up for using a style similar to the core module resolver where a `struct` is used to contain this `Vec`?

Also reaidng over this I'm not sure if the `'b` lifetime parameter throughout this module is still necessary, although I may be missing where it's needed. At least for this function though it's not required.

Reply:
I'm splitting this out to https://github.com/bytecodealliance/wasm-tools/issues/599